src/Nethermind/Nethermind.Grpc/Servers/GrpcServer.cs

```diff
@@ -61,18 +85,29 @@ public override Task<QueryResponse> Query(QueryRequest request, ServerCallContex
             }
 
             var payload = _jsonSerializer.Serialize(data);
-            _clientResults.AddOrUpdate(client ?? string.Empty, (_) =>
+            if (string.IsNullOrWhiteSpace(client))
             {
-                var results = new BlockingCollection<string>(MaxCapacity);
-                results.TryAdd(payload);
+                foreach (var (_, results) in _clientResults)
+                {
+                    try
+                    {
+                        results.TryAdd(payload);
+                    }
```

Comment:
why is it needed?

Reply:
I thought of a potential edge case - within the `Subscribe()` stream loop, in `finally` block `results` will be eventually removed, while the `PublishAsync` runs in parallel and iterates through each collection in order to add the data, but it probably needs some testing to see if it even occurs.